lib/Service/TableService.php

```diff
@@ -180,6 +180,28 @@ public function find(int $id, ?string $userId = null, bool $skipTableEnhancement
 		}
 	}
 
+	/**
+	 * @param string $term
+	 * @param int $limit
+	 * @param int $offset
+	 * @param string|null $userId
+	 * @return array
+	 */
+	public function search(string $term, int $limit = 100, int $offset = 0, ?string $userId = null): array {
+		try {
+			/** @var string $userId */
```

Comment:
Why do you redefine the type of `$userId`? It is already stated in the method signature. Also it does not match :grin:.
Plus `preCheckUserId` actually accepts a `?string` param.